qt/widgets/common/src/FunctionModel.cpp

```diff
@@ -509,14 +509,13 @@ int FunctionModel::numberOfDomains(const QList<FunctionModelDataset> &datasets)
   });
 }
 
-/// Check a domain/function index to be in range.
-void FunctionModel::checkIndex(int index) const {
-  if (index == 0)
-    return;
-  if (index < 0 || index >= getNumberDomains()) {
-    throw std::runtime_error("Domain index is out of range: " + std::to_string(index) + " out of " +
-                             std::to_string(getNumberDomains()));
-  }
+bool FunctionModel::checkIndex(int const index) const {
+  auto const indexInRange = index == 0 || (index > 0 && index < getNumberDomains());
```

Comment:
```suggestion
  auto const indexInRange = (index >= 0 && index < getNumberDomains());
```
Might be missing something, but I think it'd be fine to do this.

Reply:
I think this wouldn't work for the case where `getNumberDomains()` returns zero. We want index = 0 to be valid no matter what

Reply:
got it 👍